libcontainer/cgroups/fs2/memory.go

```diff
@@ -157,6 +158,14 @@ func getMemoryDataV2(path, name string) (cgroups.MemoryData, error) {
 	}
 	memoryData.Limit = value
 
+	// `memory.peak` since kernel 5.19
+	// `memory.swap.peak` since kernel 6.5
+	value, err = fscommon.GetCgroupParamUint(path, maxUsage)
+	if err != nil && !os.IsNotExist(err) {
+		return cgroups.MemoryData{}, err
+	}
+	memoryData.MaxUsage = value
```

Comment:
Please look into `fscommon.GetCgroupParamUnit`, If there is an error, the return value may be 0, so I think we should check err when setting `MaxUsage` value. Just like this:
```go
	value, err = fscommon.GetCgroupParamUint(path, maxUsage)
	if err != nil {
		if !os.IsNotExist(err) {
			return cgroups.MemoryData{}, err
		}
	} else {
		memoryData.MaxUsage = value
	}
```

Reply:
I think it meets the requirement: when meeting the `file not exist` error, the `MaxUsage` is set to `0`.

But it's tricky though. I will soon make a change.

Reply:
Ah, it’s not a pointer, let’s think whether we need this change or not. Sorry.